sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/math.scala

```diff
@@ -526,3 +529,171 @@ case class Logarithm(left: Expression, right: Expression)
     """
   }
 }
+
+case class Round(child: Expression, scale: Expression) extends Expression {
+
+  def this(child: Expression) = {
+    this(child, Literal(0))
+  }
+
+  override def children: Seq[Expression] = Seq(child, scale)
+
+  override def nullable: Boolean = true
+
+  override def foldable: Boolean = child.foldable
+
+  override lazy val dataType: DataType = child.dataType match {
+      case StringType | BinaryType => DoubleType
+      case DecimalType.Fixed(p, s) => DecimalType(p, _scale)
+      case t => t
+    }
+
+  override def checkInputDataTypes(): TypeCheckResult = {
+    child.dataType match {
+      case _: NumericType | NullType | BinaryType | StringType => // satisfy requirement
```

Comment:
I think `round` only makes sense for numeric type, and we should support `BinaryType` and `StringType` by adding type cast rules in `HiveTypeCoercion` or using `ExpectInputTypes`.